### DISCUSSION TITLE
Is it possible to use this with remote BT gateway?

### DISCUSSION THREAD

**Discussion by darek-margas** [tesla-local-control/tesla-local-control-addon#118] [Q&A] [answered]
I wonder what if HA runs too far or has no USB.

**Comment by iainbullock** [comment]
We don't believe BT gateway will work because the Tesla library want to have low level control of the Bluetooth device. 

For remote application I recommend using the Docker version of this project running for example on a Raspberry Pi with built in Bluetooth eg RPi3b

https://github.com/tesla-local-control/tesla_ble_mqtt_docker

**Comment by chicagoandy** [comment]
When you say 'Remote BT gateway', I take this to also mean ESPHome Bluetooth Proxies?

**Reply by iainbullock** [reply]
This won't work for the same reason as above. Direct control of the Bluetooth device is required by the Tesla library

**Comment by raphmur** [comment, marked as answer]
Indeed, BT gateway (if you mean eg. the esphome Bluetooth proxy), it will not work.
You need a fully empowered bluetooth controller.
This module is to setup directly on the Home Assistant host machine. If you cannot, then you can build a separate system and use the docker standalone variant: https://github.com/tesla-local-control/tesla_ble_mqtt_docker

**Reply by penapena** [reply]
Will it work with a Raspberry Pi Zero 2W?

**Comment by iainbullock** [comment]
It might, but I recommend you select a device with Bluetooth version>= 5.0; I believe this has Bluetooth 4.2
